### ISSUE TITLE
Set u_-1 manually

### ISSUE THREAD

**Issue by TunVunDerBunn** [do-mpc/do-mpc#84]
I have a question regarding the automatically setting of the u_-1 term in the rterm cost function part. Can this term also be set manually? I could not find anything about this in the documentation.

Thank you already in advance.

**Comment by 4flixt** [comment]
Hi @TunVunDerBunn ! So you mean the reference u to compute Delta u for the very first step of the prediction horizon? There is no direct way of setting it manually. The easiest workaround should be to create a trivial state with the dynamics:

x_k+1 = u_k (for the discrete-time case)
x_dot = u_k/dt (for the continuous-time case)

This state will thus hold the previous input and can be used to manually penalize input changes in the cost function. 
`The MPC.make_step` method needs to current initial state as an input and thus you are able to pass whatever you like for the input at u_-1.

Out of curiosity: What exactly is your use-case here? How would you set that u_-1? I am not sure if and how the documentation should be extended in this regard.

**Comment by TunVunDerBunn** [comment]
Hi @4flixt,

Thank you for the idea with the state extension, I will solve the issue like this.

I work on a project where we implemented a linear mpc which is based on a non linear model. We linearize the system at the point in the state space at which the system is at the moment. The problem is that we need to initialize a new solver at each time instant as our linear model of our system is changing over time. To be able to penialize the difference between the control input of the previous time step and the u_0 of the current time set, we need to be able to set it manually as the solver was newly initialized.

We chose this linearized approach to be able to run our controller with a higher frequency on a mircoprocessor. The do-mpc library was used to validate the controller performance in our python simulator.

Best regards,
Tun Kapgen
